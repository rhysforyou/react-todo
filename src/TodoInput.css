.todo-input {
  display: flex;
  flex-direction: row;
  align-items: baseline;
  padding: 1rem;
}

.todo-input input[type="text"] {
  flex: 1 1 auto;
  border: none;
  border-bottom: 2px solid #eee;
  padding: 5px 0;
  font-size: 100%;
  transition: border 0.2s ease-in-out;
}
.todo-input input[type="text"]:focus {
  outline: none;
  border-bottom-color: #4078c0;
}

.todo-input button {
  border: 1px solid #eee;
  border-radius: 3px;
  background-color: #FFFFFF;
  cursor: pointer;
  padding: 5px 10px;
  margin-left: 1rem;
  font-size: 100%;
}
.todo-input button:hover {
  border-color: #ccc;
}
.todo-input button:disabled {
  color: #ccc;
}
.todo-input button:disabled:hover {
  border-color: #eee;
}
