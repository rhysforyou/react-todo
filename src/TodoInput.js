import React from 'react'

const TodoInput = ({value, onChange, onSubmit}) => {
  const canSubmit = (value !== '')
  return (
    <div className='Todo-input'>
      <input type='text' value={value} onChange={onChange} />
      <input type='button' disabled={!canSubmit} value='Add Item' onClick={onSubmit} />
    </div>
  )
}
TodoInput.propTypes = {
  value: React.PropTypes.string,
  onChange: React.PropTypes.func,
  onSubmit: React.PropTypes.func
}

export default TodoInput
