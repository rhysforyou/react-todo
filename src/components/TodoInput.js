import React from 'react'
import './TodoInput.css'

const TodoInput = ({value, onChange, onSubmit}) => (
  <div className='todo-input'>
    <input type='text'
      placeholder='Add a new to-do'
      value={value}
      onChange={onChange} />
    <button disabled={value === ''} onClick={onSubmit}>Add To-Do</button>
  </div>
)
TodoInput.propTypes = {
  value: React.PropTypes.string,
  onChange: React.PropTypes.func,
  onSubmit: React.PropTypes.func
}

export default TodoInput
