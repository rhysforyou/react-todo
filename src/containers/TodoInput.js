import './TodoInput.css'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { addItem } from '../actions/todos'

export class TodoInput extends Component {
  constructor () {
    super()
    this.state = {value: ''}
  }
  handleChange (event) {
    this.setState({value: event.target.value})
  }
  handleSubmit (event) {
    event.preventDefault()
    this.props.dispatch(addItem(this.state.value))
    this.setState({value: ''})
  }
  render () {
    const canSubmit = (this.state.value !== '')
    return (
      <form className='todo-input' onSubmit={(e) => this.handleSubmit(e)}>
        <input type='text'
          placeholder='Add a new to-do'
          value={this.state.value}
          onChange={(e) => this.handleChange(e)} />
        <button type='submit' disabled={!canSubmit}>Add To-Do</button>
      </form>
    )
  }
}
TodoInput.propTypes = {
  dispatch: PropTypes.func.isRequired
}

export default connect()(TodoInput)
