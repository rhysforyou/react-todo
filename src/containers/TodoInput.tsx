import './TodoInput.css'
import React, { Component, ChangeEvent, FormEvent } from 'react'
import { connect } from 'react-redux'
import { addItem } from '../actions/todos'
import { Dispatch } from 'redux'
import { Action } from '../actions'

interface TodoInputProps {
  dispatch: Dispatch<Action>
}

interface TodoInputState {
  value: string
}

export class TodoInput extends Component<TodoInputProps, TodoInputState> {
  state: TodoInputState = {
    value: ''
  }
  handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    this.setState({ value: event.target.value })
  }
  handleSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    this.props.dispatch(addItem(this.state.value))
    this.setState({ value: '' })
  }
  render() {
    const canSubmit = this.state.value !== ''
    return (
      <form className="todo-input" onSubmit={this.handleSubmit}>
        <input
          type="text"
          placeholder="Add a new to-do"
          value={this.state.value}
          onChange={this.handleChange}
        />
        <button type="submit" disabled={!canSubmit}>
          Add To-Do
        </button>
      </form>
    )
  }
}

export default connect()(TodoInput)
