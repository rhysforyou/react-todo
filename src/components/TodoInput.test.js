/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import TodoInput from './TodoInput'

it('renders correctly', () => {
  const tree = renderer.create(
    <TodoInput value='Test Item' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
