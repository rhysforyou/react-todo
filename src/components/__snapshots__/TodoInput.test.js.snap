exports[`test renders correctly 1`] = `
<div
  className="todo-input">
  <input
    onChange={undefined}
    placeholder="Add a new to-do"
    type="text"
    value="Test Item" />
  <button
    disabled={false}
    onClick={undefined}>
    Add To-Do
  </button>
</div>
`;
